prow/pubsub/subscriber/subscriber.go

```diff
@@ -502,7 +502,26 @@ func (s *Subscriber) handleProwJob(l *logrus.Entry, jh jobHandler, msgPayload []
 		return fmt.Errorf("failed getting prowjob spec") // This should not happen
 	}
 
-	l.WithField("prowjob-event", pe).WithField("annotations", annotations).Debug("ProwjobEvent and annotations after getProwJobSpec.")
+	// Copy labels and annotations instead of modifying them, they are pointers
+	// to the static prowjobs labels and annotations.
+	combinedLabels := make(map[string]string)
+	combinedAnnotations := make(map[string]string)
+	for k, v := range labels {
+		combinedLabels[k] = v
+	}
+	for k, v := range annotations {
+		combinedAnnotations[k] = v
+	}
```

Comment:
Doing this map copy in `getProwJobSpec` instead of here would prevent this problem from occurring elsewhere in the future.

Reply:
getProwJobSpec so far is not used here so we should be fine. However it's a good point to do so, I'll do it in a separate PR later

Reply:
I will do this as part of #28036